src/gitdb.ts

```diff
@@ -20,20 +38,82 @@ class GitDB {
     return new GitDB(config);
   }
 
-  public async create(collectionName: string): Promise<string> {
-    return new Promise<string>(resolve => resolve('todo: replace logic to create collection and return its name'))
+  public async readDb(): Promise<schema> {
+    const collectionDirectoryNames = await fs.readdir(this.config.dbDir);
+
+    const collectionPromises = collectionDirectoryNames.map(
+      async (collectionName) => {
+        const collectionPath = `${this.config.dbDir}/${collectionName}`;
+
+        const documentNames = await fs.readdir(collectionPath);
+        const documentPromises = documentNames.map(async (documentName) => {
+          const documentPath = `${collectionPath}/${documentName}`;
+          const document = await fs.readFile(documentPath, 'utf8');
+          const documentData = JSON.parse(document);
+          this.collections[collectionName].insert(documentData, false);
+        });
+        await Promise.all(documentPromises);
+      }
+    );
+
+    await Promise.all(collectionPromises);
+    return this.collections;
+  }
+
+  public async createCollection(collectionName: string): Promise<string> {
+    await fse.ensureDir(this.config.dbDir);
+    this.collections[collectionName] = Collection.createCollection(
+      collectionName,
+      this
+    );
+    return collectionName;
   }
 
-  public async list(collectionName: string): Promise<string[]> {
-    return new Promise<string[]>(resolve => resolve(['todo: replace logic to return list of collection names']))
+  public async list(): Promise<string[]> {
+    const colletionNames = Object.keys(this.collections);
+    return colletionNames;
   }
 
   public async delete(collectionName: string): Promise<string> {
-    return new Promise<string>(resolve => resolve('todo: replace logic to delete collection and return its name'))
+    await fse.remove(`${this.config.dbDir}/${collectionName}`);
+    delete this.collections[collectionName];
+    return collectionName;
   }
 
-  public async commit(files: string[], message: string | undefined): Promise<string[]> {
-    return new Promise<string[]>(resolve => resolve(['todo: replace logic to return list of commited files']))
+  public async commit(
+    filePaths: string[],
+    message: string | undefined = '',
+    remove: boolean,
+    author: Author
+  ): Promise<string[]> {
+    const gitAddPromises = filePaths.map(async (filepath) => {
+      const relativeFilePath = path.relative(this.gitRoot, filepath);
+      if (!remove) {
+        await isoGit.add({
+          dir: this.gitRoot,
+          filepath: relativeFilePath,
+          fs: fsWithCallbacks
+        });
+      } else {
+        await isoGit.remove({
```

Comment:
this should better be in `then` statement instead of `else` if you convert condition to `if (remove){}`